spiceConcurrentCall_test.py

```diff
@@ -0,0 +1,109 @@
+# %%
+import spiceypy as spice # this imports the version 4.0.0
+import os
+
+# %%
+import functools
+import threading
+
+from spiceypy.spiceypy import spkezr
+
+_spicelock = threading.RLock()
+
+enablethreadinglock = False
+
+def spicelock_for_multithread(f):
+    """
+    Decorator for spiceypy to avoid concurrent calls to cspice lib.
+
+    :return:
+    """
+    @functools.wraps(f)
+    def lock(*args, **kwargs):
+        if enablethreadinglock:
+            with _spicelock:
+                try:
+                    res = f(*args, **kwargs)
+                    return res
+                except BaseException:
+                    raise
+        else:
+            try:
+                res = f(*args, **kwargs)
+                return res
+            except BaseException:
+                raise
+
+    return lock
+
+# %%
+from multiprocessing.dummy import Pool as ThreadPool
```

Comment:
I think the non-dummy threadpool should be used instead. Maybe also the regular pool to see what happens for multiple processes vs threads. In the test suite functions initialize the pools within the test functions and assume 2 threads/processes

Reply:
I used the regular ThreadPool in order to remain in the multithread environment.